CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -175,6 +181,50 @@ const char *_CFProcessPath(void) {
     }
     return __CFProcessPath;
 #else // TARGET_OS_BSD
+    char *argv0 = NULL;
+
+    // Get argv[0].
+#if defined(__OpenBSD__)
+    int mib[2] = {CTL_VM, VM_PSSTRINGS};
+    struct _ps_strings _ps;
+    size_t len = sizeof(_ps);
+
+    if (sysctl(mib, 2, &_ps, &len, NULL, 0) != -1) {
+        struct ps_strings *ps = _ps.val;
+        char *res = realpath(ps->ps_argvstr[0], NULL);
+        argv0 = res? res: strdup(ps->ps_argvstr[0]);
+    }
+#endif
+
+    if (!__CFProcessIsRestricted() && argv0 && argv0[0] == '/') {
+        _CFSetProgramNameFromPath(argv0);
+        free(argv0);
+        return __CFProcessPath;
+    }
+
+    // Search PATH.
+    if (argv0) {
+      char *paths = getenv("PATH");
+      char *p = NULL;
+      while ((p = strsep(&paths, ":")) != NULL) {
+         char pp[PATH_MAX];
+         snprintf(pp, PATH_MAX, "%s/%s", p, argv0);
+         char *res = realpath(pp, NULL);
```

Comment:
if `snprintf` returns >= `PATH_MAX` the path in `pp` will be truncated and thus invalid.

Reply:
That's fine, if the path to the binary is going to be too long, we will truncate and when we try and see if the truncated file exists, we'll get an error and we'll fall out of this case. Although, did you have an alternative suggestion in mind?

Reply:
The truncated name could still be another valid executable on the filesystem, I would suggest just treating the truncation the same way the `realpath` failure is treated.

Reply:
Good point! Fixed.